back/app/services/url_sanitization_service.rb

```diff
@@ -0,0 +1,33 @@
+# frozen_string_literal: true
+
+class UrlSanitizationService
+  private
+  class UrlScrubber < Rails::Html::PermitScrubber
+    URL_WHITELIST = [
+      %r{https:\/\/(.+)?(wistia.com|wi.st).*},
+      %r{https:\/\/.*\.typeform\.com\/to\/.*},
+      %r{https:\/\/widget\.surveymonkey\.com\/collect\/website\/js\/.*\.js},
+      %r{https:\/\/docs.google.com\/forms\/d\/e\/.*\/viewform\?embedded=true},
+      %r{https:\/\/surveys.enalyzer.com\/?\?pid=.*},
+      %r{https:\/\/www\.survey-xact\.dk\/LinkCollector\?key=.*},
+      %r{https:\/\/.*\.qualtrics\.com\/jfe\/form\/.*},
+      %r{https:\/\/www\.smartsurvey\.co\.uk\/.*},
+      %r{https:\/\/.*\.(microsoft|office)\.com\/},
+      %r{https:\/\/.*www\.eventbrite\.com\/static\/widgets\/*},
+      %r{https:\/\/.*arcgis\.com},
+      %r{https:\/\/public\.tableau\.com.*},
+      %r{https:\/\/.*datastudio\.google\.com\/embed*},
+      %r{https:\/\/app\.powerbi\.com\/},
+      %r{https:\/\/.*static\.ctctcdn\.com\/js*},
+      %r{https:\/\/.*instagram\.com},
+      %r{https:\/\/.*platform\.twitter\.com},
+      %r{https:\/\/name\.konveio\.com},
+    ].freeze
+
+    private_constant :URL_WHITELIST
+
+    def url_whitelisted?(url)
```

Comment:
@koendehondt Sorry - I maybe should have waited to make the PR. It's not quite ready for a code review. I've discussed the full test needed for all new URL options with Sebi and will be working on that this afternoon :)

Reply:
Yes I was hoping for some feedback on the regular expressions, as I'm not too familiar with writing them. Any feedback to make them more robust would be appreciated. :)

One other issue I have is that it's not always clear what kind of format the embed URLs can have from the different sources (and some are paid services, so I couldn't generate one to test). I have a feeling we might need to tweak these later depending on when customer's use them.

Reply:
The list comes from https://citizenlab.atlassian.net/browse/CL-740. @sebastienhoorens is the author of the ticket. How was the list compiled? I like to review them together, because if the regular expressions are too strict, then the user will not be able to enter some URLs. If the are too loose, then the user will be able to enter URLs that are not valid. I thought the idea was to have a list of "approved sources". If that is correct, we do not have to be too strict, but still the URLs should be valid.

Reply:
I agree that they should be the same in FE and BE, but I think they allow too many URLs. Let's talk about that today. @amanda-anderson and I will already start writing the tests.